### PR TITLE
feat: add deployment-wide DAU chart

### LINKED ISSUE #5751: Metric: Deployment-wide daily active users
Some users want to see the number of daily active users on the whole deployment, but right now we only show it by template.

Add a Metrics page to the Deployments section of the site. 

Put a DAU chart on the page.

Hook it up to a deployment DAU backend implementation.

Model it after the Template DAU metrics, but make it global to the deployment.

### PR DESCRIPTION
**Pull request by presleyp**
Closes #5751

- adds a Metrics page to Deployment Settings
- puts deployment-wide DAUs on that page

### CODE REVIEW

**Review by code-asher — APPROVED** [review]
Looks great!  The only comment I have is that it seems when there is no activity we hide the chart which makes sense to me on the templates page but I found it a little surprising to see a blank page on the deployment-wide metrics page so maybe there we could have a "no activity" message or something.

**Comment by presleyp** [comment]
> Looks great! The only comment I have is that it seems when there is no activity we hide the chart which makes sense to me on the templates page but I found it a little surprising to see a blank page on the deployment-wide metrics page so maybe there we could have a "no activity" message or something.

So glad you caught that! Ammar suggested maybe putting this in General for now since we only have one chart and I was torn but that decides it for me.

File: coderd/metricscache/metricscache.go [hunk 1]
```diff
@@ -207,6 +244,17 @@ func (c *Cache) Close() error {
...
 
+func (c *Cache) DeploymentDAUs() (*codersdk.DeploymentDAUsResponse, bool) {
+	m := c.deploymentDAUResponses.Load()
+	if m == nil {
+		// Data loading.
+		return nil, false
+	}
+
+	resp := *m
+	return &resp, true
```
Comment (ammario) [inline coderd/metricscache/metricscache.go:255]:
```suggestion
	return m, m != nil
```

**Review by ammario — DISMISSED** [review]
overall, looks good

**Comment by code-asher** [comment]
:+1: for the new location!

**Pull request merged**